### ISSUE TITLE
Remove try... convert of integers (E3002)

### ISSUE THREAD

**Issue by fatbasstard** [awslabs/cfn-python-lint#42]
In the property Type check, when the property value should be an integer (but it's not), there is an attempt to convert it:

https://github.com/awslabs/cfn-python-lint/blob/master/src/cfnlint/rules/resources/properties/Properties.py#L56

Since both YAML and JSON just handle these types out the box, the `try..except` can in theory be removed (since the type's don't match, a string is not an integer).

Personally I like the check to be more "strict', who do other think?  

Example (JSON): 
https://github.com/awslabs/cfn-python-lint/blob/master/test/templates/quickstart/vpc-management.json#L816

Example (YAML):
https://github.com/awslabs/cfn-python-lint/blob/master/test/templates/good/resource_properties.yaml#L464

These values can easily be written as actual integers
```
      - CidrIp:
          Ref: SSHLocation
        FromPort: 22 
        IpProtocol: tcp
        ToPort: 22
```

**Comment by finchd** [comment]
I actually have the reverse - an invalid integer where a String is required by CFN itself, that cfn-lint doesn't catch. The CloudFormation Designer doesn't catch it either.

In a https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticbeanstalk-configurationtemplate-configurationoptionsetting.html all properties must be Strings.
For example, an entirely reasonable-looking, but wrong use of an integer:
```json
{
                        "Namespace": "aws:autoscaling:asg",
                        "OptionName": "MinSize",
                        "Value": 12
},
```
Does anyone know if the same bit of code is related, or do I need to hunt for a bug in the Specification?

**Comment by kddejong** [comment]
@finchd and this fails when you launch to the stack?  I think it’s a similar issue to what is being described.  I think it’s a fair request by both to say that if it’s a string it’s a string and if it’s a number it’s a number and try not to translate it.   We should be capturing your issue if that stack failed to deploy because of that issue. I will need to track that down if that’s the case.  I’m not sure this is a spec issue as it looks like they use that generic property for string and number based values.

**Comment by finchd** [comment]
Yes, it definitely fails to create the stack: "Value of property Value must be of type String"
`aws cloudformation validate-template` succeeds with no errors either.

**Comment by cmmeyer** [comment]
After chatting with @kddejong tonight, I'm in the camp of strong enforcement of types.  It may trigger a lot of errors/warnings in old templates, but it's an opportunity to polish them up.  It'll be like when I first added the Flake8 linter to Atom. :)

**Comment by kddejong** [comment]
This change would catch @finchd error and satisfy this issue for @fatbasstard .  I'm good with it too.  Interesting that @finchd found a spot where integers aren't turned into strings as there are other spots where they are.   I don't want to get into the boat of writing exceptions to rules based on certain scenarios so I also like the idea of sticking to the spec the way its written.   

I have one last question for this.  This change may be annoying to some because the template could still work (in some cases).  So should we move this to a new rule with a new number?  That way people can disable it if they want.

**Comment by fatbasstard** [comment]
> I have one last question for this. This change may be annoying to some because the template could still work (in some cases). So should we move this to a new rule with a new number? That way people can disable it if they want.

They can also disable the current one, right?

Well, really love the "being strict", so 🎉 on that!

**Comment by fatbasstard** [comment]
Any change this (https://github.com/awslabs/cfn-python-lint/pull/83) is release before the next release? https://github.com/awslabs/cfn-python-lint/pull/84 will
